iceoryx_examples/request_response_in_c/client_c_waitset.c

```diff
@@ -0,0 +1,136 @@
+
+    iox_ws_storage_t waitsetStorage;
+    iox_ws_t waitset = iox_ws_init(&waitsetStorage);
+
+    iox_client_storage_t clientStorage;
+    iox_client_t client = iox_client_init(&clientStorage, "Example", "Request-Response", "Add", NULL);
+
+    if (iox_ws_attach_client_state(waitset, client, ClientState_HAS_RESPONSE, 0U, NULL) != WaitSetResult_SUCCESS)
+    {
+        printf("failed to attach client\n");
+        _exit(-1);
+    }
+
+    while (keepRunning)
+    {
+        struct AddRequest* request = NULL;
+        enum iox_AllocationResult loanResult =
+            iox_client_loan_request(client, (void**)&request, sizeof(struct AddRequest));
+        if (loanResult == AllocationResult_SUCCESS)
+        {
+            iox_request_header_t requestHeader = iox_request_header_from_payload(request);
+            iox_request_header_set_sequence_id(requestHeader, ctx.requestSequenceId);
+            ctx.expectedResponseSequenceId = ctx.requestSequenceId;
+            ctx.requestSequenceId += 1;
+            request->augend = ctx.fibonacciLast;
+            request->addend = ctx.fibonacciCurrent;
+            printf("%s Send Request: %lu %lu\n",
+                   APP_NAME,
+                   (unsigned long)ctx.fibonacciLast,
+                   (unsigned long)ctx.fibonacciCurrent);
+            iox_client_send(client, request);
+        }
+        else
+        {
+            printf("Could not allocate Request! Return value = %d\n", loanResult);
+        }
+
+        iox_notification_info_t notificationArray[NUMBER_OF_NOTIFICATIONS];
+        uint64_t missedNotifications = 0U;
+        struct timespec timeout;
+        timeout.tv_sec = 5;
+        timeout.tv_nsec = 0;
+
+        uint64_t numberOfNotifications =
+            iox_ws_timed_wait(waitset, timeout, notificationArray, NUMBER_OF_NOTIFICATIONS, &missedNotifications);
+
+        for (uint64_t i = 0; i < numberOfNotifications; ++i)
+        {
+            if (iox_notification_info_does_originate_from_client(notificationArray[0], client))
+            {
+                const struct AddResponse* response = NULL;
+                while (iox_client_take_response(client, (const void**)&response) == ChunkReceiveResult_SUCCESS)
+                {
+                    iox_const_response_header_t responseHeader = iox_response_header_from_payload_const(response);
+                    int64_t receivedSequenceId = iox_response_header_get_sequence_id_const(responseHeader);
+                    if (receivedSequenceId == ctx.expectedResponseSequenceId)
+                    {
+                        ctx.fibonacciLast = ctx.fibonacciCurrent;
+                        ctx.fibonacciCurrent = response->sum;
+                        printf("%s Got Response: %lu\n", APP_NAME, (unsigned long)ctx.fibonacciCurrent);
+                    }
+                    else
+                    {
+                        printf("Got Response with outdated sequence ID! Expected = %lu; Actual = %lu! -> skip\n",
```

Comment:
As in the basic example, this should not really happen I think. The check is alright, but I want to know whether it may legitimately happen in regular use or is just guarding against bugs/malfunctions.

Reply:
In my opinion examples should be detailed so that the user is also aware of all the little details.

Reply:
Fine with me, but it also must be clear how we intend to use it and the default use case at least should not require to much error-handling in my opinion (performance, usability and complex error-handling is ironically also error-prone). Now, this is not complex error-handling.

Ideally I would like to have a default use case with where the code is minimal and it is clear such things cannot happen (if possible, if not this is all moot). This is just my opinion: APIs should be powerful but also elegant in the default use cases.